tensorbay/opendataset/RarePlanesSynthetic/loader.py

```diff
@@ -0,0 +1,142 @@
+    from tensorbay.opendataset._utility.mocker import Image  # pylint:disable=ungrouped-imports
+try:
+    import xmltodict
+except ModuleNotFoundError:
+    from tensorbay.opendataset._utility.mocker import xmltodict  # pylint:disable=ungrouped-imports
+
+DATASET_NAME = "RarePlanesSynthetic"
+_ATTRIBUTES_GETTER: Dict[Tuple[int, ...], Callable[[str], Union[int, str, float]]] = {
+    (1, 3, 11, 13): float,
+    (14, 15): lambda x: x,
+    (12,): int,
+}
+
+
+def RarePlanesSynthetic(path: str) -> Dataset:
+    """`RarePlanesSynthetic <https://www.cosmiqworks.org/RarePlanes/>`_ dataset.
+
+    The file structure of RarePlanesSynthetic looks like::
+
+        <path>
+            images/
+                Atlanta_Airport_0_0_101_1837.png
+                ...
+            masks/
+                Atlanta_Airport_0_0_101_1837_mask.png
+                ...
+            xmls/
+                Atlanta_Airport_0_0_101_1837.xml
+                ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    category_ids = dataset.catalog.panoptic_mask.get_category_to_index()
+    segment = dataset.create_segment()
+    original_mask_dir = os.path.join(root_path, "masks")
+    new_mask_dir = os.path.join(root_path, "new_masks")
+    os.makedirs(new_mask_dir, exist_ok=True)
+    annotation_dir = os.path.join(root_path, "xmls")
+    for image_path in glob(os.path.join(root_path, "images", "*.png")):
+        segment.append(
+            _get_data(image_path, original_mask_dir, annotation_dir, new_mask_dir, category_ids)
+        )
+    return dataset
+
+
+def _get_data(
+    image_path: str,
+    original_mask_dir: str,
+    annotation_dir: str,
+    new_mask_dir: str,
+    category_ids: Dict[str, int],
+) -> Data:
+    stem = os.path.splitext(os.path.basename(image_path))[0]
+    mask_path = os.path.join(new_mask_dir, f"{stem}.png")
+    data = Data(image_path)
+    label = data.label
+    with open(os.path.join(annotation_dir, f"{stem}.xml"), "r", encoding="utf-8") as fp:
+        labels: Any = xmltodict.parse(fp.read())["image"]
+    label.box2d, label.panoptic_mask = _get_box2d_and_panoptic_mask(
+        labels["object"],
+        os.path.join(original_mask_dir, f"{stem}_mask.png"),
+        mask_path,
+        category_ids,
+    )
+    label.classification = _get_classification(labels["JSON_Variation_Parameters"]["parameter"])
+    return data
+
+
+def _get_classification(classification_labels: List[Dict[str, str]]) -> Classification:
+    classification_attributes: Dict[str, Union[int, float, str]] = {}
+    for indexes, attribute_getter in _ATTRIBUTES_GETTER.items():
+        for index in indexes:
+            attribute = classification_labels[index]
```

Comment:
is a little bit confused with variable `attribute`, `classification_attributes` and `classification_labels`